packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,255 @@
+        buttonPosition: "right",
+        majorStepSize: 10,
+        minorStepSize: 0.1,
+        stepSize: 1,
+        value: "",
+    };
+
+    private static DECREMENT_KEY = "decrement";
+    private static INCREMENT_KEY = "increment";
+
+    private static DECREMENT_ICON_NAME = "minus";
+    private static INCREMENT_ICON_NAME = "plus";
+
+    private input: HTMLInputElement;
+
+    public constructor(props?: HTMLInputProps & INumericStepperProps) {
+        super(props);
+        this.state = {
+            shouldSelectAfterUpdate: false,
+            value: props.value,
+        };
+    }
+
+    public render() {
+        const { buttonPosition, className } = this.props;
+
+        const inputGroup = this.renderInputGroup();
+        const decrementButton = this.renderButton(
+            NumericStepper.DECREMENT_KEY, NumericStepper.DECREMENT_ICON_NAME, this.handleDecrementButtonClick);
+        const incrementButton = this.renderButton(
+            NumericStepper.INCREMENT_KEY, NumericStepper.INCREMENT_ICON_NAME, this.handleIncrementButtonClick);
+
+        let elems: JSX.Element[];
+
+        if (buttonPosition === "left") {
+            elems = [decrementButton, incrementButton, inputGroup];
+        } else if (buttonPosition === "split") {
+            elems = [decrementButton, inputGroup, incrementButton];
+        } else if (buttonPosition === "right") {
+            elems = [inputGroup, decrementButton, incrementButton];
+        } else {
+            elems = [inputGroup];
+        }
+
+        const classes = classNames(Classes.CONTROL_GROUP, { [Classes.DISABLED]: this.props.disabled }, className);
+
+        return <div className={classes}>{elems}</div>;
+    }
+
+    public componentDidUpdate() {
+        if (this.state.shouldSelectAfterUpdate) {
+            this.input.setSelectionRange(0, this.state.value.length);
+        }
+    }
+
+    private renderInputGroup() {
+        return (
+            <InputGroup
+                {...removeNonHTMLProps(this.props)}
+                intent={this.props.intent}
+                inputRef={this.inputRef}
+                key="input-group"
+                onChange={this.handleInputChange}
+                onKeyDown={this.handleKeyDown}
+                value={this.state.value}
+            />
+        );
+    }
+
+    private renderButton(
+            key: string,
+            iconName: string,
+            onClick: React.EventHandler<React.MouseEvent<HTMLButtonElement>>) {
+        return (
+            <Button
+                disabled={this.props.disabled || this.props.readOnly}
+                iconName={iconName}
+                intent={this.props.intent}
+                key={key}
+                onClick={onClick}
+            />
+        );
+    }
+
+    private inputRef = (input: HTMLInputElement) => {
+        this.input = input;
+    }
+
+    private handleDecrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        const direction = -1;
```

Comment:
Mehhh okay. Also thought about making constants for this: `DIRECTION_UP` and `DIRECTION_DOWN`. But that seemed a little heavy-handed and messed with line length.